src/java/boa/compiler/visitors/CodeGeneratingVisitor.java

```diff
@@ -1386,6 +1386,12 @@ public void visit(final VarDeclStatement n) {
 			}
 		}
 
+		if (lhsType instanceof BoaArray && t instanceof BoaProtoList){
+			String[] rhsType = ((BoaProtoList)t).getType().toString().split(" ");
+			System.out.println("rhs type: " + rhsType[2]);
+			src += ".toArray(new boa.types.Code." + rhsType[2] + "[" + src + ".size()])";
```

Comment:
Nope.  We dont want to actually convert things to array!  Just pass the list around as is...

Reply:
If the list stays as it is, this error will show up in codegen instead:
`./compile/boa/ProtolistVsArray.java:139: error: incompatible types: List<CodeRepository> cannot be converted to CodeRepository[]
				___repos =  _input .getCodeRepositoriesList();`

Reply:
Right, because the codegen isnt properly handling the variable declaration.  The point of this issue was to let the compiler think protolists are an array, but not actually codegen as arrays.  They are and always will be lists (we cant control that - protocol buffers create these for us).